### PR TITLE
minihack: Add deep exploration task

### PR DESCRIPTION
**Pull request by RobertKirk**
This task has an easy solution (go to the down staircase in the starting
room) and a more difficult solution (find the room with the apples, eat
them, and then go down the staircase. The optimal policy is to do the
second solution, as the apples provide positive reward, but it's likely
that the policy will converge on going down the stair-case to avoid
per-time-step negative rewards. Hence, an algorithm that can solve this
environment should exhibit deep exploration, which often goes counter to
policy optimsation.

The environment builds off the corridor environments, but with adjust
action space, reward function and added apples.

### CODE REVIEW

**Comment by samvelyan** [comment]
Sometimes the players starts in the same room as the apples. Maybe use the same command as in level gen? Corridor envs also suffer from this occasionally.

File: nle/minihack/envs/deepexplore.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            reward=0.5,
+            repeatable=True,
+            terminal_required=False,
+            terminal_sufficient=False,
+        )
+        # Will never be achieved, but insures the environment keeps running
+        reward_manager.add_message_event(
+            ["Mission Complete"], terminal_required=True, terminal_sufficient=True
+        )
+        # Small per-timestep penalty to encourage quickly ending the episode
```
Comment (samvelyan) [inline nle/minihack/envs/deepexplore.py:30]: you could use penatly_time for this!

Comment (RobertKirk) [reply nle/minihack/envs/deepexplore.py:30]: Good point, will do

**Comment by RobertKirk** [comment]
> Sometimes the players starts in the same room as the apples. Maybe use the same command as in level gen? Corridor envs also suffer from this occasionally.

Do you mean using the BRANCH command to spawn the staicase up?

File: nle/minihack/envs/deepexplore.py (hunk 1 above)
Comment (RobertKirk) [reply nle/minihack/envs/deepexplore.py]: Have changed to use `penalty_time` kwarg

**Comment by RobertKirk** [comment]
@samvelyan I think this should be ok to merge now

**Comment by samvelyan** [comment]
Awesome, cheers!

**Pull request merged**